src/ffi.rs

```diff
@@ -382,7 +382,7 @@ pub fn view_keys(args: i32, len: i32) -> i64 {
         None => return utils::fail(),
     };
 
-    let keys: Vec<_> = (0..=MAX_KEY)
+    let keys: Vec<_> = (0..MAX_KEY)
```

Comment:
You don't need it if you have already the array as mentioned. That would reduce another loop.